### PR TITLE
feat: implement squad header changes

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Adjusted font size/colors/margins
- Smaller buttons for Public/Featured
- Added report toggle in squad menu
- Changed icon for public squads

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://mi-509-squad-header-changes.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/auth/OnboardingLogs.tsx [hunk 1]
```diff
@@ -112,6 +112,7 @@ const TiktokTracking = (): ReactElement => {
 
 interface LogSignUpProps {
   experienceLevel: keyof typeof UserExperienceLevel;
+  isExtension?: boolean;
```
Comment (rebelchris) [inline packages/shared/src/components/auth/OnboardingLogs.tsx:115]: Not sure if it's intended, but this has nothing to do with squad header right?

File: packages/shared/src/components/modals/common.tsx [hunk 2]
```diff
@@ -167,6 +167,7 @@ export const modals = {
   [LazyModal.SquadPromotion]: SquadPromotionModal,
   [LazyModal.CreateSharedPost]: CreateSharedPostModal,
   [LazyModal.ReportPost]: ReportPostModal,
+  [LazyModal.ReportSquad]: <></>,
```
Comment (rebelchris) [inline packages/shared/src/components/modals/common.tsx:170]: Modal doesn't exist yet? (If so add a todo with ticket number please)

File: packages/shared/src/components/squads/SquadHeaderMenu.tsx [hunk 3]
```diff
@@ -115,6 +116,14 @@ export default function SquadHeaderMenu({
       });
     }
 
+    list.push({
+      icon: <ContextMenuIcon Icon={FlagIcon} />,
+      action: () => {
+        openModal({ type: LazyModal.ReportSquad });
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadHeaderMenu.tsx:122]: I would think this needs to pass some props right?

File: packages/shared/src/components/squads/SquadPageHeader.tsx [hunk 4]
```diff
@@ -131,19 +147,19 @@ export function SquadPageHeader({
...
             !shouldUseListMode && MAX_WIDTH,
           )}
         >
           {squad.description}
         </p>
       )}
-      <SquadHeaderBar squad={squad} members={members} className="mt-8" />
-      <span className="mt-6 text-text-quaternary typo-footnote">
+      <SquadHeaderBar squad={squad} members={members} className="mt-5" />
+      <span className="mt-6 font-bold text-text-tertiary typo-caption1">
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadPageHeader.tsx:159]: Could move these to Typography, but optional :)

File: packages/shared/src/hooks/streaks/useStreakRecover.ts [hunk 5]
```diff
@@ -14,6 +14,7 @@ import { useToastNotification } from '../useToastNotification';
 import { useLogContext } from '../../contexts/LogContext';
 import { LogEvent, TargetType } from '../../lib/log';
 import { useAlertsContext } from '../../contexts/AlertContext';
+import { useAuthContext } from '../../contexts/AuthContext';
```
Comment (rebelchris) [inline packages/shared/src/hooks/streaks/useStreakRecover.ts:17]: Not sure why these changes are here?

File: packages/webapp/pages/onboarding.tsx [hunk 6]
```diff
@@ -185,8 +185,10 @@ export function OnboardPage(): ReactElement {
   };
 
   const onSuccessfulRegistration = (userRefetched: LoggedUser) => {
+    const isExtension = !!router.query?.install;
```
Comment (rebelchris) [inline packages/webapp/pages/onboarding.tsx:188]: Why is this here?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Could you check with design if they still want this invite button once you joined? (I don't see it in design)

Screenshot 2024-09-06 at 11 22 16

File: packages/shared/src/components/auth/OnboardingLogs.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/auth/OnboardingLogs.tsx:115]: Probably pulled from main, then pointed to this intermediate branch it is not yet updated.

File: packages/shared/src/components/modals/common.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/modals/common.tsx:170]: Let me open a draft PR to make it easy to update the intermediate branch.

File: packages/shared/src/components/modals/common/types.ts [hunk 7]
```diff
@@ -37,6 +37,7 @@ export enum LazyModal {
   SquadPromotion = 'squadPromotion',
   CreateSharedPost = 'createSharedPost',
   ReportPost = 'reportPost',
+  ReportSquad = 'reportSquad',
```
Comment (sshanzel) [inline packages/shared/src/components/modals/common/types.ts:40]: Let's use the `ReportSource`.

File: packages/shared/src/components/squads/SquadHeaderMenu.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadHeaderMenu.tsx:131]: We should not need this whole block anymore, on line 113, this was introduced.

File: packages/shared/src/components/squads/SquadPageHeader.tsx [hunk 8]
```diff
@@ -93,21 +94,36 @@ export function SquadPageHeader({
...
+                <span>Created {createdAt}</span>
+              </>
+            )}
+            {!!category && (
+              <>
+                <Separator />
+                <a
+                  aria-label={`View all squads in ${category.title}`}
+                  className="text-text-link"
+                  href={`/squads/discover/${category.title.toLowerCase()}`}
```
Comment (sshanzel) [inline packages/shared/src/components/squads/SquadPageHeader.tsx:121]: Let's use the id here. It can be found from `squad.category.id`. You might have to update the interface for Squad if it is not yet included.

File: packages/shared/src/components/squads/common/SquadPrivacyState.tsx [hunk 9]
```diff
@@ -27,15 +28,15 @@ export function SquadPrivacyState({
...
-      size={ButtonSize.Small}
+      size={ButtonSize.XSmall}
       variant={isFeatured ? ButtonVariant.Secondary : ButtonVariant.Float}
       className={classNames(
         'pointer-events-none',
         isFeatured &&
           'relative border-overlay-primary-cabbage bg-overlay-tertiary-cabbage',
       )}
     >
-      {props.copy} Squad
```
Comment (sshanzel) [inline packages/shared/src/components/squads/common/SquadPrivacyState.tsx:38]: Was the removal, intended here?

**Review by sshanzel — CHANGES_REQUESTED** [review]
Looked amazing overall. Just need to update some small details.

**Comment by sshanzel** [comment]
Also, the changes here can be shipped on its own. We can just point to main.

File: packages/shared/src/components/squads/common/SquadPrivacyState.tsx (hunk 9 above)
Comment (ilasw) [reply packages/shared/src/components/squads/common/SquadPrivacyState.tsx:38]: Yes, in design `Squad` is not mentioned anymore

**Comment by ilasw** [comment]
@sshanzel don't know if is easy to point into main, right now branch has modifies from the ga intermediate :(

**Comment by sshanzel** [comment]
> @sshanzel don't know if is easy to point into main, right now branch has modifies from the ga intermediate :(

These are just UI changes from the squad header right? It should be unrelated to the current changes in intermediate branch. Can you specify the ones you are worried about?

File: packages/shared/src/components/squads/SquadPageHeader.tsx (hunk 8 above)
Comment (ilasw) [reply packages/shared/src/components/squads/SquadPageHeader.tsx:121]: Okay, moving to id ✅

File: packages/shared/src/components/modals/common.tsx (hunk 2 above)
Comment (ilasw) [reply packages/shared/src/components/modals/common.tsx:177]: Ok removing ✅

**Review by nensidosari — APPROVED** [review]
Looks good code wise, also checked in preview.

File: packages/shared/src/components/squads/SquadPageHeader.tsx (hunk 4 above)
Comment (ilasw) [reply packages/shared/src/components/squads/SquadPageHeader.tsx]: Done!

**Comment by ilasw** [comment]
@rebelchris asket for design [here](https://dailydotdev.slack.com/archives/C069XNYMDGV/p1725887388352119).
@sshanzel moved PR to point to main

**Review by rebelchris — APPROVED** [review]

File: packages/shared/src/components/squads/SquadHeaderBar.tsx [hunk 10]
```diff
@@ -92,18 +92,19 @@ export function SquadHeaderBar({
...
           origin={Origin.SquadPage}
           size={ButtonSize.Small}
+          squad={squad}
         />
       )}
-      <SquadMemberShortList
-        squad={squad}
-        members={members}
-        size={ProfileImageSize.Small}
-      />
```
Comment (nensidosari) [inline packages/shared/src/components/squads/SquadHeaderBar.tsx:106]: Based on this [thread](https://dailydotdev.slack.com/archives/C069XNYMDGV/p1725894965874349?thread_ts=1725887388.352119&cid=C069XNYMDGV), let's revert these changes.

Comment (ilasw) [reply packages/shared/src/components/squads/SquadHeaderBar.tsx:106]: Removed! ✅

Comment (nensidosari) [reply packages/shared/src/components/squads/SquadHeaderBar.tsx]: @ilasw also the change on this file should have been reverted
